internal/provider/p2p.go

```diff
@@ -77,9 +81,9 @@ func SetupAPI(apiAddress, rootDir string, start bool, c *providerConfig.Config)
 }
 
 func SetupVPN(instance, apiAddress, rootDir string, start bool, c *providerConfig.Config) error {
-
-	if c.Kairos == nil || c.Kairos.NetworkToken == "" {
-		return fmt.Errorf("no network token defined")
+	token := ""
+	if c.P2P != nil && c.P2P.NetworkToken != "" {
+		token = c.P2P.NetworkToken
 	}
```

Comment:
Shouldn't we still fail is token is empty? I mean, we can't `SetupVPN` if there is no token, can we?

Reply:
Good point.. SetupVPN it's called also when there are VPN settings, so one could actually pass by everything via ENV and skip entirely our block - it makes it not obvious tho...